src/priceClient/adapters/acrossApi.ts

```diff
@@ -0,0 +1,43 @@
+import { msToS, PriceFeedAdapter, TokenPrice } from "../priceClient";
+import { BaseHTTPAdapter } from "./baseAdapter";
+
+type AcrossPrice = { price: number };
+const defaultTimeout = 5000; // mS
+
+export class PriceFeed extends BaseHTTPAdapter implements PriceFeedAdapter {
+  constructor(name: string, { host, timeout }: { host?: string; timeout?: number }) {
+    // Allow host to be overridden for test or alternative deployments.
+    super(name, host ?? "across.to", { timeout: timeout ?? defaultTimeout });
+  }
+
+  async getPriceByAddress(address: string, currency = "usd"): Promise<TokenPrice> {
+    const queryArgs = {
+      l1Token: address,
+      baseCurrency: currency,
+    };
+
+    const now = msToS(Date.now()) - 60; // Assume price is 60 seconds old.
+    const response: unknown = await this.query("api/coingecko", queryArgs);
+    if (!this.validateResponse(response))
+      throw new Error(`Unexpected ${this.name} response: ${JSON.stringify(response)}`);
+
+    return { address: address, price: response.price, timestamp: now };
```

Comment:
nb. Not sure if this completely hits the mark for the intended use of a predicate. At no point does the linter or tsc appear to complain about a lack of typing for response on line 24...I'm not _totally_ confident though.